fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java

```diff
@@ -124,12 +124,11 @@ public static ViewHelpers getInstance() {
         final Iterator<Triple> versions = getObjects(graph, subject, predicate);
         final Map<String, Node> map = new TreeMap<>();
         final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-        Triple triple;
-        String date;
         while (versions.hasNext()) {
-            triple = versions.next();
-            date = getVersionDate(graph, triple.getObject());
-            String key = isNullOrEmpty(date) ? format.format(new Date()) : date;
+            final Triple triple = versions.next();
+            String key = Optional.ofNullable(getVersionDate(graph, triple.getObject()))
```

Comment:
I'm more and more coming to the view that unless `Optional` is actually the type of a method return value, it should be very carefully inspected for suitability.

---
A. Soroka
The University of Virginia Library

On Jun 2, 2015, at 1:48 PM, Aaron Coburn <notifications@github.com> wrote:

> In fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java:
> 
> >          while (versions.hasNext()) {
> > -            triple = versions.next();
> > -            date = getVersionDate(graph, triple.getObject());
> > -            String key = isNullOrEmpty(date) ? format.format(new Date()) : date;
> > +            final Triple triple = versions.next();
> > +            String key = Optional.ofNullable(getVersionDate(graph, triple.getObject()))
> 
> I restored the ternary operator. I think you're right, it does read better that way
> 
> �
> Reply to this email directly or view it on GitHub.
>